sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/models/_models.py

```diff
@@ -647,9 +645,42 @@ def __init__(
     ) -> None:
         self.ack_id = ack_id
         self.error_detail = error_detail
+
+
+class SendMessageErrorOptions(_SendMessageErrorOptions):
+    """Options for send message error
+    :ivar ack_id: The ack id of the message.
+    :vartype ack_id: int
+    :ivar error_detail: The error details of the message.
+    :vartype error_detail: ~azure.messaging.webpubsubclient.models.AckMessageError
+    """
+
+    def __init__(
+        self,
+        ack_id: Optional[int] = None,
+        error_detail: Optional[AckMessageError] = None,
+    ) -> None:
+        super().__init__(ack_id, error_detail)
         self.cv = threading.Condition()
 
 
+class SendMessageErrorOptionsAsync(_SendMessageErrorOptions):
+    """Async Options for send message error
+    :ivar ack_id: The ack id of the message.
+    :vartype ack_id: int
+    :ivar error_detail: The error details of the message.
+    :vartype error_detail: ~azure.messaging.webpubsubclient.models.AckMessageError
+    """
+
+    def __init__(
+        self,
+        ack_id: Optional[int] = None,
+        error_detail: Optional[AckMessageError] = None,
+    ) -> None:
+        super().__init__(ack_id, error_detail)
+        self.event = asyncio.Event()
```

Comment:
How much are we dependent on asyncio? Is it possible to make this loop agnostic?

Reply:
SDK set assumption that users run async API in running loop as samples show and SDK uses `asyncio.event` for message synchronizntion. That's all.
Do you have any suggestions about loop agnostic?